plugins/org.polarsys.capella.scenario.editor.embeddededitor/src/org/polarsys/capella/scenario/editor/embeddededitor/commands/XtextToDiagramCommands.java

```diff
@@ -129,14 +134,17 @@ protected void doExecute() {
   }
 
   /**
-   * If a participant is in diagram, but not in editor, delete it
+   * If a participant is in diagram, but not in editor, delete it and remove all messages containing that participant
    * 
    * @param instanceRoles
    *          Instance roles from diagram
    * @param participants
    *          Participants from editor
+   * @param messagesOrReferences
+   *          All messages and references from textual editor
    */
-  private static void removeParticipantsFromDiagram(List<InstanceRole> instanceRoles, EList<Participant> participants) {
+  private static void removeDiagramParticipantsAndEditorMessages(List<InstanceRole> instanceRoles,
+      EList<Participant> participants, EList<EObject> messagesOrReferences) {
     List<InstanceRole> irToRemove = new ArrayList<InstanceRole>();
```

Comment:
you can return this list instead of upating signature of the functions.

Keep as before, the functions removeParticipantsFromDiagram with the same signature. Just return the list to be removed  (in doEditingOnParticipants, 
removedParticipants = removeParticipantsFromDiagram(...)
then call the function removeEditorMessages.

Or keep the same name on removeParticipantsFromDiagram, no need to change its name.
Add comments that for each deleted participant, we need to delete the messages in which he is involved.